server/src/test/java/org/opensearch/cluster/coordination/JoinHelperTests.java

```diff
@@ -284,7 +286,8 @@ public void testJoinFailureOnUnhealthyNodes() {
             startJoinRequest -> { throw new AssertionError(); },
             Collections.emptyList(),
             (s, p, r) -> {},
-            () -> nodeHealthServiceStatus.get()
+            () -> nodeHealthServiceStatus.get(),
+            nodeCommissioned -> {}
```

Comment:
Lets add a test that the flag in Coordinator is set as localNodeDecommissioned on JoinHelper failures and success

Reply:
Not needed anymore as `localNodeDecommissioned` is removed